shuup/core/api/front_products.py

```diff
@@ -0,0 +1,607 @@
+
+    def get_identifier(self, product_attribute):
+        return product_attribute.attribute.identifier
+
+
+class CategorySerializer(serializers.ModelSerializer):
+    image = serializers.SerializerMethodField()
+
+    def get_image(self, category):
+        if category.image:
+            return self.context["request"].build_absolute_uri(category.image.url)
+
+    class Meta:
+        model = Category
+        fields = ("id", "name", "slug", "image")
+
+
+class CompleteShopProductSerializer(serializers.ModelSerializer):
+    product_id = serializers.ReadOnlyField()
+    name = serializers.ReadOnlyField(source='product.name')
+    short_description = serializers.ReadOnlyField(source='product.short_description')
+    description = serializers.ReadOnlyField(source='product.description')
+    image = serializers.SerializerMethodField()
+    categories = CategorySerializer(many=True)
+    attributes = serializers.SerializerMethodField()
+    price = serializers.SerializerMethodField()
+    net_weight = serializers.ReadOnlyField(source='product.net_weight')
+    sales_unit = serializers.SerializerMethodField()
+    is_orderable = serializers.SerializerMethodField()
+    cross_sell = serializers.SerializerMethodField()
+    variations = serializers.SerializerMethodField()
+    package_content = serializers.SerializerMethodField()
+    shop = ShopSerializer()
+
+    class Meta:
+        model = ShopProduct
+        fields = [
+            "id",
+            "product_id",
+            "name",
+            "short_description",
+            "description",
+            "image",
+            "categories",
+            "attributes",
+            "price",
+            "net_weight",
+            "variations",
+            "shop",
+            "is_orderable",
+            "sales_unit",
+            "cross_sell",
+            "package_content",
+        ]
+
+    def _get_pricing_context(self, request, shop):
+        return PricingContext(shop=shop, customer=self.context["customer"])
+
+    def get_image(self, shop_product):
+        image = shop_product.product.primary_image
+        if not image:
+            return
+
+        if image.external_url:
+            return image.external_url
+        elif image.file:
+            return self.context["request"].build_absolute_uri(image.file.url)
+
+    def get_attributes(self, shop_product):
+        return AttributeSerializer(shop_product.product.attributes, many=True).data
+
+    def get_price(self, shop_product):
+        key, val = context_cache.get_cached_value(identifier="shop_product_price",
+                                                  item=shop_product,
+                                                  context={},
+                                                  allow_cache=True)
+        if val is not None:
+            return val
+
+        context = self._get_pricing_context(self.context["request"], shop_product.shop)
+        price = shop_product.product.get_price(context).value
```

Comment:
maybe the `.value` here is not correct. We should also return it since the `base_price` should be in the product `PriceInfo`.